### ISSUE TITLE
Support `xarray.DataArray` from the Xarray library

### ISSUE THREAD

**Issue by alshan** [JetBrains/lets-plot#739]
from [discussion](https://github.com/JetBrains/lets-plot/discussions/611#discussioncomment-5361403)

> (https://docs.xarray.dev/en/stable/user-guide/data-structures.html). Xarray is a very powerful library that can be used to create dataframe for 2D grids. It is very similar to Pandas and if Lets-Plot could (maybe it already can, but I have not succeeded yet) handle dictionary of grids the way it handles Dataframes it could be used to handle xarray objects, which I think could benefit Lets-Plot.
There is one library that handle xarray object in a interactive way, it is hvplot (https://hvplot.holoviz.org/). For exemple it can generate a figure showing different grids using an interactive slider.

**Comment by alshan** [comment]
There is a new Python package - [tidydraws](https://drbenvincent.github.io/tidydraws/). As far as I understand, it can flatten multi-dimensional formats (ArviZ / xarray) into long-form DataFrames (Polars/Pandas). @egayer , would this help?

**Comment by egayer** [comment]
Yes, I see how that works, and it does, in fact, partially address my original request. However, I realize just how redundant this request is. I was specifically referring to the ability to facet grids using `geom_imshow()`, just as `xarray` already does (https://docs.xarray.dev/en/latest/user-guide/plotting.html). I understand that this isn’t essential for ‘lets-plot’ and I won’t dwell on it, given that there’s an alternative.
Thank you for giving it some thought!
